api/src/services/HierarchyCollector.ts

```diff
@@ -78,21 +73,19 @@ class HierarchyCollector {
         if (datastreams.includes("OCR-DIRTY")) {
             extraDetails.fullText.ocrDirty = [await this.fedora.getDatastreamAsString(pid, "OCR-DIRTY")];
         }
-        const models = relations.hasModel ?? [];
-        if (models.includes("info:fedora/vudl-system:DOCData") || models.includes("info:fedora/vudl-system:PDFData")) {
+        if (models.includes("vudl-system:DOCData") || models.includes("vudl-system:PDFData")) {
             const extractor = new TikaExtractor((await this.fedora.getDatastream(pid, "MASTER")).body, this.config);
             extraDetails.fullText.fromDocument = [extractor.extractText()];
         }
         return extraDetails;
     }
 
-    async getHierarchy(pid: string, fetchRdf = true): Promise<FedoraData> {
-        const result = await this.getFedoraData(pid, fetchRdf);
+    async getHierarchy(pid: string): Promise<FedoraData> {
+        const result = await this.getFedoraData(pid);
         // Create promises to retrieve parents asynchronously...
-        const promises = (result.relations.isMemberOf ?? []).map(async (resource) => {
-            const parentPid = resource.substr("info:fedora/".length);
-            // The "false" here skips RDF retrieval:
-            const parent = await this.getHierarchy(parentPid, false);
+        const promises = (result.fedoraDetails.isMemberOf ?? []).map(async (resource) => {
```

Comment:
What happens when one of these promises fail? Do you have a try catch for this Promise.all?

Reply:
At this point, all the error handling is occurring in the Express.js routes -- all calls (indirect or direct) to getHierarchy are wrapped in try..catch at the route layer so that appropriate error codes can be returned from the framework if something goes wrong.